### PR TITLE
Add L1 traffic test(s)

### PR DESCRIPTION
**Pull request by StepanVovkPLV**
Depends on [PR](https://github.com/PLVision/dent-testing/pull/62), and [PR](https://github.com/PLVision/dent-testing/pull/47)

### CODE REVIEW

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/constants.py [hunk 1]
```diff
@@ -106,5 +107,6 @@
     "suite_group_functional": [
         "suite_functional_ipv4",
         "suite_functional_bridging",
+        "suite_functional_l1"
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/constants.py]:
```suggestion
        "suite_functional_l1",
```

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+@pytest.mark.parametrize("speed , duplex",
+                         [
+                            # (10, "full"),
+                            # (10, "half"),
+                            # (100, "full"),
+                            # (100, "half")
+                            # (1000, "full"),
+                            # (1000, "half"),
+                            (10000, "full"),
+                         ])
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]:
```suggestion
@pytest.mark.parametrize(
    "speed, duplex",
    [# (10, "full"), (10, "half"),
     # (100, "full"), (100, "half"),
     # (1000, "full"), (1000, "half"),
     (10000, "full")],
)
```

Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: How are you going to test 10/100/1000 speed?

Comment (SerhiyBoikoPLV) [reply DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: Another issue is that running each test would take a long time. Consider testing both full and half duplex during a single test run (10 full+half, 100 full+half, 1000 full, 10_000 full)

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        "device": port,
+        "operstate": "up",
+        "master": bridge
+    } for port in ports]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Failed setting link to state UP."
+
+    # 4. Set up port duplex, speed
+    out = await Ethtool.set(input_data=[{device_host_name: [{
+        "devname": port,
+        "speed": 10000,
```
Comment (SerhiyBoikoPLV) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: use a variable

**Review by SerhiyBoikoPLV — CHANGES_REQUESTED** [review]
Please make sure that you are restoring the default speed/duplex after the test (by using a fixture or a try-catch-finally)

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        print.error("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    bridge = "br0"
+
+    # Check dut product name. If it supports the speed parameter
+    model = await OnlpSystemInfo.show(input_data=[{device_host_name: [{}]}], parse_output=True)
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: Please move device type recognition to the fixture and skip the test if it is not applicable.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    6. Set up traffic according to duplex mode
+    7. Send traffic to rx_ports
+    8. Verify received traffic on ports is as expected
+    9. Transmitted rate equals received rate (and matches them configuration)
+    """
+
+    # 1. Retrieve tgen and dent device info
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print.error("The testbed does not have enough dent with tgen connections")
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: What is this? There is no such attribute in print.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    bridge = "br0"
+
+    # Check dut product name. If it supports the speed parameter
+    model = await OnlpSystemInfo.show(input_data=[{device_host_name: [{}]}], parse_output=True)
+    switch = model[0][device_host_name]['parsed_output']['product_name']
+    if "AS5114" not in switch and speed < 1000:
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: I suggest adding some external (e.g. YAML) file with models and port types. Please do not hardcode model names in test's code.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+                            (10000, "full"),
+                         ])
+async def test_l1_forced_speed_10_full_duplex(testbed, speed, duplex):
+    """
+    Test Name: L1: forced speed
+    Test Suite: suite_functional_bridging
+    Test Overview: Verify speed, duplex configuration with packet forwarding
+    Test Procedure:
+    1. Retrieve tgen and dent device info
+    2. Create bridge entity
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: Do you relly need a bridge in this test scenario?

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+    # 6. Set up traffic according to duplex mode
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    # TODO Add support for hals duplex streams
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: The most TODO here is to switch TG to a required speed/duplex mode.

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+       }})
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 7. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(1)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    await tgen_utils_start_traffic(tgen_dev)
```
Comment (anton7811) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_forced_speed.py]: Why do you send traffic two times?

**Review by anton7811 — CHANGES_REQUESTED** [review]

File: DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_autodetect.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+        "duplex": duplex} for port in ports]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Failed setting port duplex, speed."
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24}])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+    #  Configure ixia ports
+    await tgen_utils_update_l1_config(tgen_dev, tg_ports, speed=speed, autoneg=False)
+    await asyncio.sleep(60)  # wait needed in case port was down before
```
Comment (yuriilisovskyi) [inline DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_autodetect.py]: Why do you wait 60 seconds here, but in other tests 20 seconds?

**Review by yuriilisovskyi — APPROVED** [review]

**Pull request closed without merging**